src/foam/nanos/dig/SugarWebAgent.java

```diff
@@ -123,19 +141,59 @@ public void execute(X x) {
               }
 
               paramTypes[j] = pArray[j].getType();
-              arglist[j] = p.getParameter(pArray[j].getName());
+
+              if ( p.getParameter(pArray[j].getName()) != null ) {
+                // the get method
+                arglist[j] = p.getParameter(pArray[j].getName());
+              } else {
+                // the post method
+                paramName = pArray[j].getName();
+              }
 
               logger.debug(pArray[j].getName() + " :   " + p.getParameter(pArray[j].getName()));
 
               // casting and setting according to parameters type
               String typeName = pArray[j].getType().getCanonicalName();
-              Class paramObj_ = null;
-              Object obj_ = null;
 
               if ( typeName.equals("foam.core.X") ) {
                 arglist[j] = x;
               } else if ( ! SafetyUtil.isEmpty(p.getParameter(pArray[j].getName())) ) {
+                //get method
                 arglist[j] = setParameterValue(x, resp, out, p, typeName, pArray[j]);
+              } else if ( paramName != null ) {
+                //post method
+                //TODO array
+                if ( typeName.equals("double") || typeName.equals("boolean") || typeName.equals("int") || typeName.equals("long") || typeName.equals("long") || typeName.equals("java.lang.String") ) {
```

Comment:
You shouldn't do it this way. You can add all of the supported types to a Set (which could be made a constant), then do a Set.contains here instead of having this long if statement.